dot/node_integration_test.go

```diff
@@ -0,0 +1,355 @@
+	err = keystore.LoadKeystore("alice", ks.Gran)
+	require.NoError(t, err)
+	require.Equal(t, 1, ks.Gran.Size())
+	err = keystore.LoadKeystore("alice", ks.Babe)
+	require.NoError(t, err)
+	require.Equal(t, 1, ks.Babe.Size())
+
+	cfg.Core.Roles = types.AuthorityRole
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	bp := node.Services.Get(&babe.Service{})
+	require.NotNil(t, bp)
+	fg := node.Services.Get(&grandpa.Service{})
+	require.NotNil(t, fg)
+}
+
+func TestStartNode(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+	require.NotNil(t, genFile)
+
+	cfg.Init.Genesis = genFile
+	cfg.Core.GrandpaAuthority = false
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	err = keystore.LoadKeystore("alice", ks.Gran)
+	require.NoError(t, err)
+	err = keystore.LoadKeystore("alice", ks.Babe)
+	require.NoError(t, err)
+
+	cfg.Core.Roles = types.FullNodeRole
+
+	node, err := NewNode(cfg, ks)
+	require.NoError(t, err)
+
+	go func() {
+		<-node.started
+		node.Stop()
+	}()
+
+	err = node.Start()
+	require.NoError(t, err)
+}
+
+func TestInitNode_LoadGenesisData(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+	require.NotNil(t, genPath)
+
+	cfg.Init.Genesis = genPath
+	cfg.Core.GrandpaAuthority = false
+
+	err := InitNode(cfg)
+	require.NoError(t, err)
+
+	config := state.Config{
+		Path: cfg.Global.BasePath,
+	}
+	stateSrvc := state.NewService(config)
+
+	gen, err := genesis.NewGenesisFromJSONRaw(genPath)
+	require.NoError(t, err)
+
+	genTrie, err := genesis.NewTrieFromGenesis(gen)
+	require.NoError(t, err)
+
+	genesisHeader, err := types.NewHeader(common.NewHash([]byte{0}), genTrie.MustHash(), trie.EmptyHash,
+		big.NewInt(0), types.NewDigest())
+	require.NoError(t, err)
+
+	err = stateSrvc.Initialise(gen, genesisHeader, genTrie)
+	require.NoError(t, err)
+
+	err = stateSrvc.Start()
+	require.NoError(t, err)
+
+	defer func() {
+		err = stateSrvc.Stop()
+		require.NoError(t, err)
+	}()
+
+	gendata, err := stateSrvc.Base.LoadGenesisData()
+	require.NoError(t, err)
+
+	testGenesis := NewTestGenesis(t)
+
+	expected := &genesis.Data{
+		Name:       testGenesis.Name,
+		ID:         testGenesis.ID,
+		Bootnodes:  common.StringArrayToBytes(testGenesis.Bootnodes),
+		ProtocolID: testGenesis.ProtocolID,
+	}
+	require.Equal(t, expected, gendata)
+
+	genesisHeader, err = stateSrvc.Block.BestBlockHeader()
+	require.NoError(t, err)
+
+	stateRoot := genesisHeader.StateRoot
+	expectedHeader, err := types.NewHeader(common.NewHash([]byte{0}), stateRoot, trie.EmptyHash, big.NewInt(0),
+		types.NewDigest())
+	require.NoError(t, err)
+	require.Equal(t, expectedHeader.Hash(), genesisHeader.Hash())
+}
+
+func TestInitNode_LoadStorageRoot(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genPath := NewTestGenesisAndRuntime(t)
+	require.NotEmpty(t, genPath)
```

Comment:
nit do we need this check? Aren't we testing the test code here 😄

Reply:
removed.